app/js/controllers/foyer/recapSituation.js

```diff
@@ -155,56 +153,20 @@ angular.module('ddsCommon').controller('RecapSituationCtrl', function($scope, $s
             $scope.hasRessources = true;
         }
 
-        $scope.ressourcesTns = [];
-        $scope.ressourcesNonTns = [];
+        $scope.ressources = [];
         ressourceTypes.forEach(function(ressourceType) {
             if ($scope.tempRessources[ressourceType.id]) {
-                $scope.ressourcesNonTns.push({
+                var ressource = {
                     type: ressourceType,
-                    totalMensuel: $scope.tempRessources[ressourceType.id].totalMensuel,
                     totalAnnuel: Math.round($scope.tempRessources[ressourceType.id].totalAnnuel)
-                });
-            }
-        });
-
-        var ressourcesMicroFiscal = {
-            type: _.find(ressourceTypes, { id: 'caMicroEntreprise' }),
-            totalAnnuel: 0
-        };
-        var ressourcesAutresTns = {
-            type: _.find(ressourceTypes, { id: 'autresRevenusTns' }),
-            totalAnnuel: 0
-        };
-        var ressourceAgricoleTns = {
-            type: _.find(ressourceTypes, { id: 'revenusAgricolesTns' }),
-            totalAnnuel: 0
-        };
-        $scope.situation.individus.forEach(function(individu) {
-            if (individu.caMicroEntreprise) {
-                ressourcesMicroFiscal.totalAnnuel += individu.caMicroEntreprise;
-                $scope.hasRessources = true;
-            }
-            if (individu.autresRevenusTns) {
-                ressourcesAutresTns.totalAnnuel += individu.autresRevenusTns;
-                $scope.hasRessources = true;
-            }
-           if (individu.revenusAgricolesTns) {
-                ressourceAgricoleTns.totalAnnuel += individu.revenusAgricolesTns;
-                $scope.hasRessources = true;
+                };
+                var ressourcesWithoutMonthlyAmount = ['caMicroEntreprise','revenusAgricolesTns','autresRevenusTns'];
+                if (! _.contains(ressourcesWithoutMonthlyAmount, ressourceType.id)) {
```

Comment:
Searching for a given specific string in an array works as well with the native API, and will be faster and decrease your dependency on external libraries: [`ressourcesWithoutMonthlyAmount.indexOf(resourceType.id) < 0`](https://developer.mozilla.org/fr/docs/Web/JavaScript/Reference/Objets_globaux/Array/indexOf).